app/src/main/java/com/example/androidapp/DatabaseController.java

```diff
@@ -167,6 +173,75 @@ public static void updateUser(Context context, User user) {
 
     }
 
+
+    public static void insertStep(String s, String day, String hour, Context context) {
+        ContentValues values = new ContentValues();
+        values.put("timestamp", s);
+        values.put("day", day);
+        values.put("hour", hour);
+        DatabaseController databaseHelper = new DatabaseController(context);
+        SQLiteDatabase database = databaseHelper.getReadableDatabase();
+        database.insert("Step",null,values);
+    }
+
+    public static void loadSteps(Context context){
+        List<String> dates = new LinkedList<String>();
+        DatabaseController databaseHelper = new DatabaseController(context);
+        SQLiteDatabase database = databaseHelper.getReadableDatabase();
+
+        Cursor cursor = database.query("Step", new String [] {"timestamp"}, null, null, "timestamp",
+                null, null );
+
+        cursor.moveToFirst();
+        for (int index=0; index < cursor.getCount(); index++){
+            dates.add(cursor.getString(0));
+            cursor.moveToNext();
+        }
+        database.close();
+
+    }
+
+
+    public static Integer loadSingleStep(Context context, String date){
```

Comment:
I am a bit confused by this method: the signature suggests that it returns one single step given a day in input (which doesn't make sense given that the user makes more than one step per day) but the implementation calculates the total sum of steps in that day.

So the name of the method have to be changed and I suggest to write also a comment to describe what it does.

Reply:
true, totally wrong name but the method is correct right?